<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from wagon_V3.xacro                 | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="wagon" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <material name="silver">
    <color rgba="0.700 0.700 0.700 1.000"/>
  </material>
  <transmission name="Rev1_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev1_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev2_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev2_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev3_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev3_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev4_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev4">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev4_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev5_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev5">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev5_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev6_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev6">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev6_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="Rev7_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="Rev7">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="Rev7_actr">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="ros_control">
      <!--<robotNamespace>/summit_xl</robotNamespace>-->
      <!--https://answers.ros.org/question/292444/gazebo_ros_control-plugin-gazeboroscontrolplugin-missing-legacymodens-defaultrobothwsim/-->
      <robotParam>cart_description</robotParam>
      <!-- <controlPeriod>0.001</controlPeriod> -->
      <!-- <robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType> -->
      <legacyModeNS>true</legacyModeNS>
    </plugin>
  </gazebo>
  <gazebo reference="base_link">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
    <gravity>true</gravity>
  </gazebo>
  <gazebo reference="Wheel_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="Wheel_2">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="Wheel_3">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="Wheel_4">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="Handle_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="Wheel_Mount_Swivel_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="Wheel_Mount_Swivel_2">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="Wheel_Mount_Fixed_2">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="Wheel_Mount_Fixed_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="pocket_1">
    <material>Gazebo/Silver</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>true</selfCollide>
  </gazebo>
  <link name="root_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <material name="silver"/>
    </visual>
  </link>
  <link name="base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0.1773025237734832 0.14164285524102913 -0.458084473687539"/>
      <mass value="14.332500497057262"/>
      <inertia ixx="1.815445" ixy="0.0" ixz="0.0" iyy="2.191965" iyz="0.011056" izz="0.900189"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Wheel_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.017999759836384868 4.7540047370953076e-08 -4.889678157093957e-07"/>
      <mass value="5.4476936846398445"/>
      <inertia ixx="0.023058" ixy="0.0" ixz="0.0" iyy="0.012" iyz="-0.0" izz="0.012"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.069241 0.139414 0.071657"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.069241 0.139414 0.071657"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Wheel_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.017999759836385576 4.754004720441962e-08 -4.88967815820418e-07"/>
      <mass value="5.4476936846398445"/>
      <inertia ixx="0.023058" ixy="0.0" ixz="0.0" iyy="0.012" iyz="-0.0" izz="0.012"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.386759 0.139414 0.071657"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.386759 0.139414 0.071657"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Wheel_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.017999523773476106 4.7540046926863866e-08 -4.889678172359524e-07"/>
      <mass value="5.4476936846398445"/>
      <inertia ixx="0.023058" ixy="-0.0" ixz="-0.0" iyy="0.012" iyz="-0.0" izz="0.012"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.005797 0.139414 0.834457"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.005797 0.139414 0.834457"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_3.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Wheel_4">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.018000476226523787 1.6154440396792813e-05 1.1109012534427087e-05"/>
      <mass value="5.4476936846398445"/>
      <inertia ixx="0.023058" ixy="-0.0" ixz="0.0" iyy="0.012" iyz="-0.0" izz="0.012"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.360403 0.139414 0.834457"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_4.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.360403 0.139414 0.834457"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_4.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Handle_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-4.762265238822394e-07 0.43470096474489145 3.112531270967403e-05"/>
      <mass value="4.648080140534756"/>
      <inertia ixx="0.273315" ixy="0.0" ixz="0.0" iyy="0.002739" iyz="6.7e-05" izz="0.275703"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.177303 0.015064 0.024157"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Handle_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.177303 0.015064 0.024157"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Handle_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Wheel_Mount_Swivel_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.07233701445022406 0.20736342377329953 0.03700153648198488"/>
      <mass value="0.4390624166336173"/>
      <inertia ixx="0.000873" ixy="-3e-06" ixz="-0.0" iyy="0.00017" iyz="-0.000158" izz="0.00085"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.123395 0.255978 0.073315"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_Mount_Swivel_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.123395 0.255978 0.073315"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_Mount_Swivel_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Wheel_Mount_Swivel_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.07233701445022428 0.20736342377329964 0.03700153648198488"/>
      <mass value="0.4390624166336173"/>
      <inertia ixx="0.000873" ixy="-3e-06" ixz="-0.0" iyy="0.00017" iyz="-0.000158" izz="0.00085"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.332605 0.255978 0.073315"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_Mount_Swivel_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.332605 0.255978 0.073315"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_Mount_Swivel_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Wheel_Mount_Fixed_2">
    <inertial>
      <origin rpy="0 0 0" xyz="-4.7622652393775056e-07 -0.041655358517784354 -4.918676156373536e-07"/>
      <mass value="0.21332050921311865"/>
      <inertia ixx="0.000356" ixy="-0.0" ixz="0.0" iyy="9.1e-05" iyz="-0.0" izz="0.000387"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.342403 0.029564 0.834457"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_Mount_Fixed_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.342403 0.029564 0.834457"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_Mount_Fixed_2.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="Wheel_Mount_Fixed_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-4.762265238163199e-07 -0.04165535851778424 -4.918676153042867e-07"/>
      <mass value="0.21332050921311865"/>
      <inertia ixx="0.000356" ixy="-0.0" ixz="-0.0" iyy="9.1e-05" iyz="0.0" izz="0.000387"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.012203 0.029564 0.834457"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_Mount_Fixed_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.012203 0.029564 0.834457"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/Wheel_Mount_Fixed_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="pocket_1">
    <inertial>
      <origin rpy="0 0 0" xyz="-4.76226525020218e-07 -0.26343800902767844 -0.3768754918676184"/>
      <mass value="185.3138608124999"/>
      <inertia ixx="10.321663" ixy="0.0" ixz="0.0" iyy="13.570381" iyz="0.0" izz="4.635436"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="-0.177303 -0.313336 0.052157"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/pocket_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="silver"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.177303 -0.313336 0.052157"/>
      <geometry>
        <mesh filename="package://wagon_description/meshes/pocket_1.stl" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="Rev1" type="continuous">
    <origin rpy="0 0 0" xyz="0.054154 0.116564 0.001658"/>
    <parent link="Wheel_Mount_Swivel_1"/>
    <child link="Wheel_1"/>
    <dynamics damping="0.1000"/>
    <dynamics friction="0.0000"/>
    <axis xyz="1.0 0.0 0.0"/>
  </joint>
  <joint name="Rev2" type="continuous">
    <origin rpy="0 0 0" xyz="0.054154 0.116564 0.001658"/>
    <parent link="Wheel_Mount_Swivel_2"/>
    <child link="Wheel_2"/>
    <dynamics damping="0.1000"/>
    <dynamics friction="0.0000"/>
    <axis xyz="1.0 0.0 0.0"/>
  </joint>
  <joint name="Rev3" type="continuous">
    <origin rpy="0 0 0" xyz="-0.018 -0.10985 0.0"/>
    <parent link="Wheel_Mount_Fixed_1"/>
    <child link="Wheel_3"/>
    <dynamics damping="0.1000"/>
    <dynamics friction="0.0000"/>
    <axis xyz="1.0 0.0 0.0"/>
  </joint>
  <joint name="Rev4" type="continuous">
    <origin rpy="0 0 0" xyz="0.018 -0.10985 0.0"/>
    <parent link="Wheel_Mount_Fixed_2"/>
    <child link="Wheel_4"/>
    <dynamics damping="0.1000"/>
    <dynamics friction="0.0000"/>
    <axis xyz="1.0 0.0 0.0"/>
  </joint>
  <joint name="Rev5" type="revolute">
    <origin rpy="0 0 0" xyz="0.177303 -0.015064 -0.024157"/>
    <parent link="base_link"/>
    <child link="Handle_1"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="100" lower="0.0" upper="0.523599" velocity="100"/>
  </joint>
  <joint name="Rev6" type="continuous">
    <origin rpy="0 0 0" xyz="-0.123395 -0.255978 -0.073315"/>
    <parent link="base_link"/>
    <child link="Wheel_Mount_Swivel_1"/>
    <dynamics damping="0.1000"/>
    <dynamics friction="0.0000"/>
    <axis xyz="0.0 1.0 0.0"/>
  </joint>
  <joint name="Rev7" type="continuous">
    <origin rpy="0 0 0" xyz="0.332605 -0.255978 -0.073315"/>
    <parent link="base_link"/>
    <child link="Wheel_Mount_Swivel_2"/>
    <dynamics damping="0.1000"/>
    <dynamics friction="0.0000"/>
    <axis xyz="0.0 1.0 0.0"/>
  </joint>
  <joint name="Rigid8" type="fixed">
    <origin rpy="0 0 0" xyz="0.342403 -0.029564 -0.834457"/>
    <parent link="base_link"/>
    <child link="Wheel_Mount_Fixed_2"/>
  </joint>
  <joint name="Rigid9" type="fixed">
    <origin rpy="0 0 0" xyz="0.012203 -0.029564 -0.834457"/>
    <parent link="base_link"/>
    <child link="Wheel_Mount_Fixed_1"/>
  </joint>
  <joint name="Rigid11" type="fixed">
    <origin rpy="0 0 0" xyz="0.177303 0.313336 -0.052157"/>
    <parent link="base_link"/>
    <child link="pocket_1"/>
  </joint>
  <joint name="Rigid10" type="fixed">
    <origin rpy="1.57079632679 0 1.57079632679" xyz="-0.177303 0 0.23"/>
    <parent link="root_link"/>
    <child link="base_link"/>
  </joint>
</robot>

